src/enhance-collection/__tests__/index.js

```diff
@@ -32,9 +40,63 @@ test("statinamic/lib/enhance-collection", (t) => {
       }
     ),
     [
-      { k: "ey" },
-      { k: "ay" },
+      {
+        k: "ey",
+        l: "hi",
+      },
+      {
+        k: "ay",
+        q: "hu"
+      },
     ],
     "should filter by object { key: regexp }"
   )
 })
+
+test("filter by custom function", (t) => {
+  t.same(
+    enhanceCollection(
+      collec,
+      {
+        filter: (t) => t.k === "eye",
+      }
+    ),
+    [
+      { k: "eye" },
+    ]
+  )
+})
+
+test("filter by string", (t) => {
+  t.same(
+    enhanceCollection(
+      collec,
+      {
+        filter: "l",
+      }
+    ),
+    [
+      {
+        k: "ey",
+        l: "hi",
+      }
+    ],
+  )
+})
+
+test("multiple filters", (t) => {
```

Comment:
yeah? What can be the issue?
We should also test that the array can be object or function too.